src/transaction/log_page_buffer.c

```diff
@@ -1663,33 +1682,33 @@ logpb_copy_page (THREAD_ENTRY * thread_p, LOG_PAGEID pageid, LOG_CS_ACCESS_MODE
       goto exit;
     }
 
-  log_bufptr = &log_Pb.buffers[logpb_get_log_buffer_index ((int) pageid)];
+  index = logpb_get_log_buffer_index ((int) pageid);
+  if (index >= 0 && index < log_Pb.num_buffers)
+    {
+      log_bufptr = &log_Pb.buffers[index];
+    }
+  else
+    {
+      er_set (ER_FATAL_ERROR_SEVERITY, ARG_FILE_LINE, ER_LOG_PAGE_CORRUPTED, 1, pageid);
+      return NULL;
```

Comment:
I have changed it. Do you have another observations or it is everything clear now?